libafl_frida/src/asan/asan_rt.rs

```diff
@@ -483,40 +485,55 @@ impl AsanRuntime {
     #[allow(clippy::too_many_lines)]
     pub fn register_hooks(&mut self, gum: &Gum) {
         let mut interceptor = Interceptor::obtain(gum);
-
         macro_rules! hook_func {
             ($lib:expr, $name:ident, ($($param:ident : $param_type:ty),*), $return_type:ty) => {
                 paste::paste! {
                     log::trace!("Hooking {}", stringify!($name));
 
                     let target_function = frida_gum::Module::find_export_by_name($lib, stringify!($name)).expect("Failed to find function");
-                    self.hooks.insert(stringify!($name).to_string(), target_function);
+
+                    static mut [<$name:snake:upper _PTR>]: *const c_void = 0 as *const c_void;
```

Comment:
Weren't we going to have the static mut be transmuted already to the target function pointer type?

Reply:
Rust would complain when I initialized a static function pointer to 0, so I was forced to do it this way.

Reply:
can we make this a non-mut static? declare and initialize in one step? use -1 as the uninitialized value? It would be much better to have the function pointer type...